src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarm.java

```diff
@@ -488,7 +488,35 @@ public ListSuitesResult listSuites(String runArn) {
         return api.listSuites(request);
     }
 
+    public int getUnmeteredDevices(String os) {
+        AccountSettings accountSettings = getAccountSettings();
+        if (accountSettings == null) {
+            return 0;
+        } else if (os.equalsIgnoreCase("ANDROID")) {
+            return getAccountSettings().getUnmeteredDevices().get("ANDROID");
+        } else if (os.equalsIgnoreCase("IOS")) {
+            return getAccountSettings().getUnmeteredDevices().get("IOS");
+        } else {
+            return 0;
+        }
+    }
 
+    public String getOs(String appArtifact) {
+        if (appArtifact.toLowerCase().endsWith("apk")) {
+            return "Android";
+        } else {
+            return "IOS";
+        }
+    }
+
+    public AccountSettings getAccountSettings() {
+        try {
+            GetAccountSettingsRequest request = new GetAccountSettingsRequest();
+            return api.getAccountSettings(request).getAccountSettings();
```

Comment:
api.getAccountSettings(request)  will never return a null. I checked the code.
We construct getAccountSettingresult using new and then fill it